pkg/ingester/ingester_test.go

```diff
@@ -5677,3 +5677,123 @@ func TestGetIgnoreSeriesLimitForMetricNamesMap(t *testing.T) {
+		oooFirstSample int64
+		oooLastSample  int64
+		expPushError   bool
+	}{
+		"should return in order and out of order data": {
+			queryFrom:      math.MinInt64,
+			queryTo:        math.MaxInt64,
+			oooFirstSample: 70 * time.Minute.Milliseconds(),
+			oooLastSample:  99 * time.Minute.Milliseconds(),
+			expPushError:   false,
+		},
+		"if ooo samples go back past allowance writing should return an error": {
+			queryFrom:      math.MinInt64,
+			queryTo:        math.MaxInt64,
+			oooFirstSample: 10 * time.Minute.Milliseconds(), // Allowance is 30 min, first sample is at minute 100 so first ooo sample is too old
+			oooLastSample:  99 * time.Minute.Milliseconds(),
+			expPushError:   true,
+		},
+	}
+
+	// Configure ingester with OOO Allowance
+	cfg := defaultIngesterTestConfig(t)
+	cfg.BlocksStorageConfig.TSDB.OOOAllowance = 30 * time.Minute
+	cfg.BlocksStorageConfig.TSDB.OOOCapMin = 4
+	cfg.BlocksStorageConfig.TSDB.OOOCapMax = 32
+
+	// Create ingester
+	i, err := prepareIngesterWithBlocksStorage(t, cfg, nil)
+	require.NoError(t, err)
+	require.NoError(t, services.StartAndAwaitRunning(context.Background(), i))
+	defer services.StopAndAwaitTerminated(context.Background(), i) //nolint:errcheck
+
+	// Wait until it's healthy
+	test.Poll(t, 1*time.Second, 1, func() interface{} {
+		return i.lifecycler.HealthyInstancesCount()
+	})
+
+	ctx := user.InjectOrgID(context.Background(), "test")
+
+	// Push first in order sample at minute 100
+	firstInOrderSample := 100 * time.Minute.Milliseconds()
+	serie := series{
+		lbls:      labels.Labels{{Name: labels.MetricName, Value: "test_1"}, {Name: "status", Value: "200"}},
+		value:     float64(firstInOrderSample),
+		timestamp: firstInOrderSample,
+	}
+
+	req, _, _, _ := mockWriteRequest(t, serie.lbls, serie.value, serie.timestamp)
+	_, err = i.Push(ctx, req)
+	require.NoError(t, err)
+
+	// Run tests
+	for testName, testData := range tests {
+		t.Run(testName, func(t *testing.T) {
+			expSamples := []model.SamplePair{
+				{
+					Timestamp: model.Time(firstInOrderSample),
+					Value:     model.SampleValue(firstInOrderSample),
+				},
+			}
+
+			// Push ooo samples within the allowance time in a single request
+			var samples []mimirpb.Sample
+			var lbls []labels.Labels
+			for j := testData.oooFirstSample / time.Minute.Milliseconds(); j < testData.oooLastSample/time.Minute.Milliseconds(); j++ {
+				min := int64(j) * time.Minute.Milliseconds()
+				samples = append(samples, mimirpb.Sample{
+					TimestampMs: min,
+					Value:       float64(min),
+				})
+				lbls = append(lbls, serie.lbls)
+				expSamples = append(expSamples, model.SamplePair{
+					Timestamp: model.Time(min),
+					Value:     model.SampleValue(min),
+				})
+			}
+			wReq := mimirpb.ToWriteRequest(lbls, samples, nil, nil, mimirpb.API)
+			_, err = i.Push(ctx, wReq)
+			if testData.expPushError {
+				require.Error(t, err, "should have failed on push")
+				require.ErrorAs(t, err, &storage.ErrTooOldSample)
+			} else {
```

Comment:
One way of reducing indentation of some code is to have return for the smaller block and avoid the `else` block. With this you can get rid of the `else` block.
```suggestion
			if testData.expPushError {
				require.Error(t, err, "should have failed on push")
				require.ErrorAs(t, err, &storage.ErrTooOldSample)
				return
			}
```